File: Views/Onboarding/ListeningStepView.swift
import SwiftUI

struct ListeningStepView: View {
    @ObservedObject var viewModel: OnboardingViewModel
    
    private let listeningOptions = [
        "listening_most_of_time",
        "listening_sometimes",
        "listening_rarely"
    ]
    
    var body: some View {
        VStack(spacing: 0) {
            // Espace entre barre de progression et titre (harmonisé)
            Spacer()
                .frame(height: 40)
            
            // Titre centré à gauche
            HStack {
                Text("listening_question".localizedOnboarding)
                    .font(.system(size: 28, weight: .bold))
                    .foregroundColor(.black)
                    .multilineTextAlignment(.leading)
                Spacer()
            }
            .padding(.horizontal, 30)
            
            // Sous-titre explicatif
            HStack {
                Text("private_answer_note".localizedOnboarding)
                    .font(.system(size: 14, weight: .regular))
                    .foregroundColor(.black.opacity(0.6))
                    .multilineTextAlignment(.leading)
                Spacer()
            }
            .padding(.horizontal, 30)
            .padding(.top, 8)
            
            Spacer()
            
            // Options de réponse
            VStack(spacing: 12) {
                ForEach(Array(listeningOptions.enumerated()), id: \.offset) { index, localizationKey in
                    Button(action: {
                        print("🔥 ListeningStepView: Option sélectionnée: \(localizationKey)")
                        viewModel.listeningAnswer = localizationKey
                    }) {
                        Text(localizationKey.localizedOnboarding)
                            .font(.system(size: 16))
                            .foregroundColor(viewModel.listeningAnswer == localizationKey ? .white : .black)
                            .multilineTextAlignment(.leading)
                            .lineLimit(nil)
                            .fixedSize(horizontal: false, vertical: true)
                            .frame(maxWidth: .infinity, alignment: .leading)
                            .padding(.vertical, 16)
                            .padding(.horizontal, 20)
                            .background(
                                RoundedRectangle(cornerRadius: 12)
                                    .fill(viewModel.listeningAnswer == localizationKey ? Color(hex: "#FD267A") : Color.white)
                                    .overlay(
                                        RoundedRectangle(cornerRadius: 12)
                                            .stroke(
                                                viewModel.listeningAnswer == localizationKey ? Color(hex: "#FD267A") : Color.black.opacity(0.1),
                                                lineWidth: viewModel.listeningAnswer == localizationKey ? 2 : 1
                                            )
                                    )
                                    .shadow(color: Color.black.opacity(0.05), radius: 8, x: 0, y: 4)
                            )
                    }
                    .buttonStyle(PlainButtonStyle())
                }
            }
            .padding(.horizontal, 30)
            
            Spacer()
            
            // Zone blanche collée en bas avec bouton "Continuer"
            VStack(spacing: 0) {
                Button(action: {
                    print("🔥 ListeningStepView: Bouton continuer pressé")
                    viewModel.nextStep()
                }) {
                    Text("continue".localized)
                        .font(.system(size: 18, weight: .semibold))
                        .foregroundColor(.white)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 16)
                        .background(Color(hex: "#FD267A"))
                        .cornerRadius(28)
                        .opacity(viewModel.listeningAnswer.isEmpty ? 0.5 : 1.0)
                }
                .disabled(viewModel.listeningAnswer.isEmpty)
                .padding(.horizontal, 30)
            }
            .padding(.vertical, 30)
            .background(Color.white)
        }
        .background(Color(red: 0.97, green: 0.97, blue: 0.98))
    }
}

#Preview {
    ListeningStepView(viewModel: OnboardingViewModel())
}

File: Views/Onboarding/ListeningStepView.kt
// ListeningStepScreen.kt
package com.yourapp.onboarding

import android.util.Log
import androidx.annotation.StringRes
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewmodel.compose.viewModel

// Couleurs (équivalents de ton Swift)
private val Pink = Color(0xFFFD267A)
private val LightBg = Color(0xFFF7F7FA)
private val BorderDefault = Color.Black.copy(alpha = 0.10f)

// Option avec clé (persistée dans le ViewModel) + resId (affichage localisé)
private data class ListeningOption(
    val key: String,
    @StringRes val labelRes: Int
)

@Composable
fun ListeningStepScreen(
    viewModel: OnboardingViewModel = viewModel()
) {
    val tag = "ListeningStepScreen"
    val options = remember {
        listOf(
            ListeningOption("listening_most_of_time", R.string.listening_most_of_time),
            ListeningOption("listening_sometimes",   R.string.listening_sometimes),
            ListeningOption("listening_rarely",      R.string.listening_rarely)
        )
    }

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(LightBg)
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
        ) {
            Spacer(Modifier.height(40.dp))

            // Titre
            Row(
                modifier = Modifier.padding(horizontal = 30.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = stringResource(R.string.listening_question),
                    fontSize = 28.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.Black
                )
                Spacer(Modifier.weight(1f))
            }

            // Sous-titre
            Row(
                modifier = Modifier
                    .padding(horizontal = 30.dp)
                    .padding(top = 8.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = stringResource(R.string.private_answer_note),
                    fontSize = 14.sp,
                    fontWeight = FontWeight.Normal,
                    color = Color.Black.copy(alpha = 0.6f)
                )
                Spacer(Modifier.weight(1f))
            }

            Spacer(Modifier.height(24.dp))

            // Options
            Column(
                modifier = Modifier.padding(horizontal = 30.dp),
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                options.forEachIndexed { index, option ->
                    val selected = viewModel.listeningAnswer == option.key
                    OptionCard(
                        text = stringResource(option.labelRes),
                        selected = selected,
                        onClick = {
                            Log.d(tag, "🔥 Option sélectionnée: ${option.key} (index=$index)")
                            viewModel.listeningAnswer = option.key
                        }
                    )
                }
            }

            Spacer(Modifier.weight(1f))
            // Espace pour ne pas être masqué par la zone blanche du bas
            Spacer(Modifier.height(120.dp))
        }

        // Zone blanche collée en bas + bouton Continuer
        Column(
            modifier = Modifier
                .align(Alignment.BottomCenter)
                .fillMaxWidth()
                .background(Color.White)
                .padding(vertical = 30.dp)
        ) {
            val enabled = viewModel.listeningAnswer.isNotEmpty()
            Button(
                onClick = {
                    Log.d(tag, "🔥 Bouton continuer pressé")
                    viewModel.nextStep()
                },
                enabled = enabled,
                modifier = Modifier
                    .padding(horizontal = 30.dp)
                    .fillMaxWidth()
                    .alpha(if (enabled) 1f else 0.5f),
                colors = ButtonDefaults.buttonColors(
                    containerColor = Pink,
                    contentColor = Color.White
                ),
                shape = RoundedCornerShape(28.dp)
            ) {
                // NOTE: si ta clé est exactement "continue", R.string.`continue` (backticks) est requis en Kotlin
                Text(
                    text = stringResource(R.string.`continue`),
                    fontSize = 18.sp,
                    fontWeight = FontWeight.SemiBold
                )
            }
        }
    }
}

@Composable
private fun OptionCard(
    text: String,
    selected: Boolean,
    onClick: () -> Unit
) {
    val bg = if (selected) Pink else Color.White
    val fg = if (selected) Color.White else Color.Black
    val borderColor = if (selected) Pink else BorderDefault
    val borderWidth = if (selected) 2.dp else 1.dp

    Box(
        modifier = Modifier
            .fillMaxWidth()
            .shadow(elevation = 8.dp, shape = RoundedCornerShape(12.dp), clip = false)
            .background(bg, RoundedCornerShape(12.dp))
            .border(borderWidth, borderColor, RoundedCornerShape(12.dp))
            .padding(vertical = 16.dp, horizontal = 20.dp)
            .noRippleClickable(onClick),
    ) {
        Text(
            text = text,
            fontSize = 16.sp,
            color = fg
        )
    }
}

/**
 * Extension utilitaire pour un clic sans effet ripple
 * (équivalent du PlainButtonStyle() côté SwiftUI).
 */
import androidx.compose.foundation.clickable
import androidx.compose.ui.composed

private fun Modifier.noRippleClickable(onClick: () -> Unit): Modifier = composed {
    this.then(
        Modifier.clickable(
            indication = null,
            interactionSource = remember { androidx.compose.foundation.interaction.MutableInteractionSource() }
        ) { onClick() }
    )
}

/* ---------- Exemple de ViewModel minimal ----------
   Adapte-le si tu as déjà un OnboardingViewModel côté Android.
*/
class OnboardingViewModel : ViewModel() {
    var listeningAnswer by mutableStateOf("")
    fun nextStep() {
        // TODO: navigation / logique suivante
    }
}
